boomerang.js

```diff
@@ -369,780 +363,986 @@ boomr = {
+			// These are expected for cross-origin iframe access, although the Internet Explorer check will only
+			// work for browsers using English.
+			return err.name === "SecurityError" ||
+				(err.name === "TypeError" && err.message === "Permission denied") ||
+				(err.name === "Error" && err.message && err.message.match(/^(Permission|Access is) denied/));
+		},
 
-	addError: function(err, src, extra) {
-		var str;
-		if (typeof err !== "string") {
-			str = String(err);
-			if(str.match(/^\[object/)) {
-				str = err.name + ": " + (err.description || err.message).replace(/\r\n$/, "");
+		addVar: function(name, value) {
+			if (typeof name === "string") {
+				impl.vars[name] = value;
 			}
-			err = str;
-		}
-		if (src !== undefined) {
-			err = "[" + src + ":" + BOOMR.now() + "] " + err;
-		}
-		if (extra) {
-			err += ":: " + extra;
-		}
+			else if (typeof name === "object") {
+				var o = name, k;
+				for (k in o) {
+					if (o.hasOwnProperty(k)) {
+						impl.vars[k] = o[k];
+					}
+				}
+			}
+			return this;
+		},
 
-		if (impl.errors[err]) {
-			impl.errors[err]++;
-		}
-		else {
-			impl.errors[err] = 1;
-		}
-	},
+		removeVar: function(arg0) {
+			var i, params;
+			if (!arguments.length) {
+				return this;
+			}
 
-	addVar: function(name, value) {
-		if(typeof name === "string") {
-			impl.vars[name] = value;
-		}
-		else if(typeof name === "object") {
-			var o = name, k;
-			for(k in o) {
-				if(o.hasOwnProperty(k)) {
-					impl.vars[k] = o[k];
+			if (arguments.length === 1
+					&& Object.prototype.toString.apply(arg0) === "[object Array]") {
+				params = arg0;
+			}
+			else {
+				params = arguments;
+			}
+
+			for (i=0; i<params.length; i++) {
+				if (impl.vars.hasOwnProperty(params[i])) {
+					delete impl.vars[params[i]];
 				}
 			}
-		}
-		return this;
-	},
 
-	removeVar: function(arg0) {
-		var i, params;
-		if(!arguments.length) {
 			return this;
-		}
+		},
 
-		if(arguments.length === 1
-				&& Object.prototype.toString.apply(arg0) === "[object Array]") {
-			params = arg0;
-		}
-		else {
-			params = arguments;
-		}
+		hasVar: function(name) {
+			return impl.vars.hasOwnProperty(name);
+		},
+
+		requestStart: function(name) {
+			var t_start = BOOMR.now();
+			BOOMR.plugins.RT.startTimer("xhr_" + name, t_start);
 
-		for(i=0; i<params.length; i++) {
-			if(impl.vars.hasOwnProperty(params[i])) {
-				delete impl.vars[params[i]];
+			return {
+				loaded: function(data) {
+					BOOMR.responseEnd(name, t_start, data);
+				}
+			};
+		},
+
+		responseEnd: function(name, t_start, data) {
+			if (typeof name === "object" && name.url) {
+				impl.fireEvent("xhr_load", name);
 			}
-		}
+			else {
+				BOOMR.addVar("xhr.pg", name);
+				BOOMR.plugins.RT.startTimer("xhr_" + name, t_start);
+				impl.fireEvent("xhr_load", {
+					"name": "xhr_" + name,
+					"data": data
+				});
+			}
+		},
+
+		/**
+		 * Undo XMLHttpRequest instrumentation and reset the original
+		 */
+		uninstrumentXHR: function() {
```

Comment:
if instrumentXHR and uninstrumentXHR are implemented in auto_xhr.js why is this here?